server/queues/processors/NotificationsProcessor.ts

```diff
@@ -72,6 +74,13 @@ export default class NotificationsProcessor extends BaseProcessor {
       const notify = await this.shouldNotify(document, recipient.user);
 
       if (notify) {
+        const notification = await Notification.create({
+          event: event.name,
+          userId: recipient.user.id,
+          actorId: document.updatedBy.id,
+          teamId: team.id,
+          documentId: document.id,
+        });
         await DocumentNotificationEmail.schedule({
```

Comment:
What do you think about passing `metadata` as second argument instead of mixing in with email props?

Reply:
Hmm...makes sense from the POV of _decoupling_ it from email notifications. It might be needed for other types of notifications(in-app etc...) as well. What's your reasoning?

Reply:
Reasoning is mainly they are functionally different data use, the first being for template injection.